### PR TITLE
MIDI-22: Prepare an ECG segmentation dataset

### PR DESCRIPTION
**Pull request by JasiekKaczmarczyk**
[Prepare an ECG segmentation dataset](https://sacrebleu.atlassian.net/browse/MIDI-22)
Script for processing data

Visualization of how processed dataset looks like. I used window of 1000 samples which is approximately 8 second signal.
Blue - ECG signal (one of channels)
Orange - mask which will be used for segmentation (area of each beat is 13 samples which corresponds to approximately 100 ms)
subset_with_mask

Here is sample processed file. It has three attributes: "channel_1" and "channel_2" which will be used to predict "mask" attribute.
[file-00-indices-136001-137000.csv](https://github.com/Nospoko/ecg-segmentation-unet/files/12290773/file-00-indices-136001-137000.csv)

### CODE REVIEW

**Comment by roszcz** [comment]
Signal processing works great 👍 

For the dataset I think we should try to avoid dealing with multiple csv files. Here's a mock interface I had in mind for the huggingface dataset building pipeline:

```python
from datasets import Dataset

def process_record(record_path: str) -> list[dict]:
  ...
  records = []
  for subset in tqdm(data.rolling(window=sequence_window, step=sequence_window)):
      record = {
        "signal": subset[["channel_1", "channel_2"]].values,
        "mask": subset.mask.values,
      }
      records.append(record)
  return records

records = []
for record_path in ltafdb_paths:
  records += process_record(record_path)

dataset = Dataset.from_list(records)
dataset.push_to_hub(dataset_name, token)
```

Let me know if this makes sense - to test you can run it on a single record to see how it performs.

You can also look into the ways to define data types in HF datasets, something like this could be useful, and can potentially improve the iteration performance 10-100x:

```python
features = Features({
        'signal': Array2D(dtype="float", shape=(1000, 2)),
        'mask': Array2D(dtype="float", shape=(1000, 1)),
})
dataset = dataset.cast(features)
```

FYI I copied this from another project, so don't expect it to work as is - first thing to look into is if we should use Array1D for the mask field.

**Comment by JasiekKaczmarczyk** [comment]
I updated prepare_dataset.py to create huggingface dataset and push it into the hub.
[Dataset url](https://huggingface.co/datasets/JasiekKaczmarczyk/physionet-preprocessed)

**Comment by roszcz** [comment]
I ran this on all of the records, and created a clone of your dataset https://huggingface.co/datasets/roszcz/ecg-segmentation-ltafdb, signal processing took less than 15 minutes, so it's very manageable.

File: requirements.txt
Comment (roszcz) [inline requirements.txt:1]: I think it's missing `datasets` :)
Also, can you pin exact versions of all the libraries in the `requirements.txt`?

**Review by roszcz — APPROVED** [review]

File: requirements.txt
Comment (JasiekKaczmarczyk) [reply requirements.txt:1]: Ok, I'll update that

**Pull request merged**